commit 8e0e1af77a3823e819f314be530b5d362698f0a2
Author: Jonas Maebe <jonas.maebe@elis.ugent.be>
Date:   Wed Dec 17 13:52:24 2014 +0100

    Reset $a/$d state also when switching back to a previously encountered section

diff --git a/gas/config/obj-elf.c b/gas/config/obj-elf.c
index 6d48a94..f131c5f 100644
--- a/gas/config/obj-elf.c
+++ b/gas/config/obj-elf.c
@@ -519,6 +519,9 @@ get_section (bfd *abfd ATTRIBUTE_UNUSED, asection *sec, void *inf)
 	      && strcmp (group_name, gname) == 0));
 }
 
+/* Used to mark data section for DIABLO */
+extern int prev_was_data;
+
 /* Handle the .section pseudo-op.  This code supports two different
    syntaxes.
 
@@ -577,6 +580,9 @@ obj_elf_change_section (const char *name,
     {
       sec = old_sec;
       subseg_set (sec, 0);
+
+      /* Regenerate a $d or $a for Diablo */
+      prev_was_data = -1;
     }
   else
     sec = subseg_force_new (name, 0);
